internal/service/alarms/internal/resourceserver/resourceserver.go

```diff
@@ -0,0 +1,122 @@
+package resourceserver
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/oapi-codegen/oapi-codegen/v2/pkg/securityprovider"
+
+	"github.com/openshift-kni/oran-o2ims/internal/controllers/utils"
+	"github.com/openshift-kni/oran-o2ims/internal/service/alarms/internal/resourceserver/generated"
+	"github.com/openshift-kni/oran-o2ims/internal/service/common/clients"
+)
+
+const (
+	resourceServerURLEnvName = "RESOURCE_SERVER_URL"
+	tokenPathEnvName         = "TOKEN_PATH"
+)
+
+type Resource = generated.Resource
+type ResourceType = generated.ResourceType
+
+type ResourceServer struct {
+	client        *generated.ClientWithResponses
+	Resources     *[]Resource
+	ResourceTypes *[]ResourceType
+}
+
+// New creates a new resource server object
+func New() (*ResourceServer, error) {
+	url := utils.GetServiceURL(utils.InventoryResourceServerName)
+
+	// Use for local development
+	resourceServerURL := os.Getenv(resourceServerURLEnvName)
+	if resourceServerURL != "" {
+		url = resourceServerURL
+	}
+
+	// Set up transport
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{
+			MinVersion: tls.VersionTLS12,
+			// #nosec G402: InsecureSkipVerify set true.
+			InsecureSkipVerify: utils.GetTLSSkipVerify(),
+		},
+	}
+
+	if !tr.TLSClientConfig.InsecureSkipVerify {
+		tr.TLSClientConfig.RootCAs = x509.NewCertPool()
+		data, err := os.ReadFile(utils.DefaultServiceCAFile)
+		if err != nil {
+			// Should never happen
+			return nil, fmt.Errorf("failed to read CA file: %w", err)
+		}
+
+		tr.TLSClientConfig.RootCAs.AppendCertsFromPEM(data)
+	}
+
+	hc := http.Client{Transport: tr}
```

Comment:
We have a utility to do this. See `GetDefaultTLSConfig` in utils

Reply:
or `GetDefaultBackendTransport` depending on what you need

Reply:
I'm using `GetDefaultBackendTransport` now. Thanks